### ISSUE TITLE
aria-expanded is not announced in Chrome

### ISSUE THREAD

**Issue by nousrnm** [nvaccess/nvda#11277]
Please download the code. This is not a duplicate of #9096. The button doesn't have an aria-haspopup attribute.

https://drive.google.com/file/d/1ZfpmAJz_-swwqdlwUYGipCYS7dSK74n2/view?usp=sharing

### Steps to reproduce:
Open the sub navigation: https://youtu.be/ixtKsrX_-pc
Close the sub navigation by pressing Enter when focus is on MEN: https://youtu.be/i_VyP_3Tc1Y
Close the sub navigation by pressing Tab + Shift when focus is on MEN: https://youtu.be/CixqtdL5QM4

### Actual behavior:
Nothing is announced.

### Expected behavior:
NVDA should announce "Expanded" or "Collapsed" based on the new value.

When you open the sub navigation: https://youtu.be/R_iXH78cC9Y
When you close the sub navigation by pressing Enter when focus is on MEN: https://youtu.be/sd9NSX-jcsc
When you close the sub navigation by pressing Tab + Shift when focus is on MEN: https://youtu.be/nxFXkGH03iw

#### NVDA installed/portable/running from source:
Installed.

#### NVDA version:
2020.1

#### Windows version:
10 Home, Version 2004.

#### Name and version of other software in use when reproducing the issue:
Google Chrome, Version 83.0.4103.106 (Official Build) (64-bit)

#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.

#### If addons are disabled, is your problem still occuring?
I don't have any addons.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No.

[SCREENSHOT UPDATED] If I remove these parts of the code, NVDA behaves as expected, so maybe it's a Chrome bug...
https://drive.google.com/file/d/15knDXhOXxh80IwyJ8XIbxRKAaPHkhwD5/view?usp=sharing

**Comment by Adriani90** [comment]
Oh sorry, I couldn't see the code. Could you please copy and paste the code snippet to this issue which should be removed in order for expanded state to be reported correctly? Unfortunately you linked a png file with the code snippet. But many developers of NVDA are blind or visually impaired, so they might not be able to see the code in the image.
cc: @feerrenrut, @ObjectInSpace

**Comment by nousrnm** [comment]
Sure.

The parts of the code that prevent NVDA from working properly in Chrome (and in all the other browsers based on Blink I assume) are:

This one, located in the openSubNavigation(e) function:
`this.subNavigationTogglers[this.subNavigationId].lastElementChild.removeAttribute("aria-hidden");`

And this one, located in the closeSubNavigation() function:
`this.subNavigationTogglers[this.subNavigationId].lastElementChild.setAttribute("aria-hidden", true);`

**Comment by feerrenrut** [comment]
Any chance you could make a minimal example to demonstrate this issue? At the moment the example is quite large and contains a lot of unnecessary code.

**Comment by nousrnm** [comment]
@feerrenrut 

Yes.
I will be back in a few hours.

**Comment by nousrnm** [comment]
@feerrenrut 

Done:
https://drive.google.com/file/d/12Fks22kz2pNI5wHodM4YUCF0YVmOXHst/view?usp=sharing

**Comment by nousrnm** [comment]
The example in the link above is a little different from the example in the main post. Here is how to reproduce the issue in the minimal version of the code:

1. Move focus on the "Catalog" button and press enter.
What happens when you do that? NVDA doesn't say anything.
What should have happened? NVDA should have said "Expanded".

2. Move focus on the "Catalog" button press "Enter", and hit "Enter" again while focus is on the "Catalog" button.
What happens when you do that? NVDA doesn't say anything.
What should have happened? NVDA should have said "Collapsed".

3. Move focus on the "Catalog" button press "Enter", then press Tab + Shift while focus is on the "Catalog" button.
What happens when you do that? NVDA doesn't say anything.
What should have happened? NVDA should have said "Collapsed".

**Comment by ObjectInSpace** [comment]
I wasn't able to get either of these to work in Firefox or JAWS either. Firefox wouldn't even rout focus the buttons. Are you sure the examples are correct?

**Comment by nousrnm** [comment]
@ObjectInSpace 

Thank you for trying.
Yes, I think I used the aria attributes correctly (if that's what you mean). I tested the content of 
page-minimal.zip in Firefox with NVDA and works as it should.

**Comment by JulienCochuyt** [comment]
@nousrnm, the provided example does not work well on my side either with Chrome 83 or Firefox 76: I get very different results depending on screen size.

Still, why toggle `aria-hidden` `true`/removed?
Sure, `aria-hidden="false"` is strongly discouraged, but I'd bet dynamically setting/removing it as implemented in the provided example is not much widely supported either.

This attribute really means that the content of the so tagged sub tree is either redundant or extraneous.
Reference: https://www.w3.org/TR/wai-aria-1.1/#aria-hidden
To dynamically control visibility, you have other means like the HTML `hidden` attribute, or CSS `display` and `visibility` styles.
Toggling these is much widely supported.

By the way, I wouldn't advise to play with dynamically toggling `tabindex` `-1`/removed either, which might lead to inconsistent results across browsers.

Please note I'm not arguing that Chrome or NVDA shouldn't enhance their flexibility towards eg. `aria-hiden="false"`.
My point is just that your use of the attribute is IMHO rather borderline in the provided example.
Btw., I didn't check which of Chrome or NVDA is blocking your scenario here. All apologies  if I'm missing the point.

**Comment by nousrnm** [comment]
> To dynamically control visibility, you have other means like the HTML hidden attribute, or CSS display and visibility styles.
> Toggling these is much widely supported.
> 

Toggling these is bad for performance...

**Comment by leonardder** [comment]
I can sometimes reproduce this, but not always. Are you sure this is an issue in NVDA and not in Chrome itself? I'm tempted to think the latter.

**Comment by nousrnm** [comment]
> Are you sure this is an issue in NVDA and not in Chrome itself?

No. I thought you could tell me...

**Comment by nousrnm** [comment]
@leonardder 
> Are you sure this is an issue in NVDA and not in Chrome itself?

No. I thought you could tell me...

**Comment by Adriani90** [comment]
The minimal test case is not available for download anymore. Could some one upload the file to this issue please? Ogherwise it will be really hard to investigate further. @nousrnm could you please test again with last NVDA alpha version? There were some changes which might have improved the behavior here.

**Comment by feerrenrut** [comment]
None of the links function on this issue, the example was never attached to the issue directly. There is nothing we can do with this. Closing the issue. @nousrnm If you are still interested in getting this resolved please open a new issue, in future please attach examples as txt, zip, or simply paste into the description.
